### PR TITLE
Dockerfile rework and id remapping

### PR DESCRIPTION
**Pull request by StaticRocket**
Reformat the Dockerfile, use entrypoint ID remapping, clean up pip artifacts, fix git submodule/worktree mapping, remove proxies.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -123,18 +131,32 @@ RUN apt-get update \
...
-		lld
+		lld \
+	&& echo "**** cleanup ****" \
+	&& apt-get autoremove \
+	&& apt-get clean \
+	&& rm -rf \
+		/tmp/* \
+		/var/lib/apt/lists/* \
+		/var/tmp/* \
+		/var/log/*
```
[the remark this answers is not in the document]
Comment (StaticRocket) [reply Dockerfile]: Not easily without inadvertently committing that script or adding an additional mount point for each `RUN` block that interacts with apt.

File: Dockerfile [hunk 2]
```diff
@@ -105,6 +107,18 @@ RUN test "$INSTALL_GCC" -eq "1" \
 
 COPY other-configs/ /
 
+RUN echo "**** create abc user and make our folders ****" \
+	&& useradd -u 1000 -U -d /config -s /bin/false abc \
```
[the remark this answers is not in the document]
Comment (StaticRocket) [reply Dockerfile:111]: The `abc` user is more common in community id remapping containers but sure.

File: Dockerfile [hunk 3]
```diff
@@ -87,6 +87,22 @@ RUN apt-get update \
 		/var/tmp/* \
 		/var/log/*
 
+# install toolchains if requested
+RUN test "$INSTALL_GCC" -eq "1" \
+	&& apt-get update \
+	&& apt-get install -y --no-install-recommends \
+		gcc-aarch64-linux-gnu \
```
[the remark this answers is not in the document]
Comment (StaticRocket) [reply Dockerfile:96]: Hehe, we could switch this base out for arch and solve this fairly easily...

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (StaticRocket) [reply Dockerfile]: Well, that's not entirely true. You can technically define an `ARG` to hold some shell script and execute it like a function but that's gross.

File: Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (nmenon) [reply Dockerfile:96]: Either we use kernel.org compiler or we use the ARM compiler. I am not in favor of using the distro cross compiler as they tend to stale up rather fast.

[the remark this answers is not in the document]
Comment (StaticRocket) [reply Dockerfile:96]: You say that but I'm pretty sure this image will be stale before the arch toolchains are.

Regardless, I'll drop the offending change for now.

**Pull request merged**